cbmc/proofs/proof_guide.md

```diff
@@ -0,0 +1,391 @@
+# CBMC Proof Guide and Cookbook for MLKEM-C
```

Comment:
Maybe put this into `cbmc/` as `README.md`? Or at least reference it from there? Otherwise people might not find it, while I imagine a few people will curiously open `cbmc/` and stumble into whatever README is there.

Reply:
Well... I put it in cbmc/proofs because it's a proof-related artefact, but yes - let's link to it from the main README. I will take a look.

Reply:
I have added comments and links in the top-level README and in cbmc/proofs/README.md
See what you think.